mobile/src/main/java/org/openhab/habdroid/util/CacheManager.java

```diff
@@ -33,7 +33,13 @@ public static CacheManager getInstance(Context context) {
 
     private CacheManager(Context appContext) {
         mHttpCache = new Cache(new File(appContext.getCacheDir(), "http"), 2 * 1024 * 1024);
-        mBitmapCache = new LruCache<HttpUrl, Bitmap>(2048) {
+
+        // Get max available VM memory, exceeding this amount will throw an
+        // OutOfMemory exception. Stored in kilobytes as LruCache takes an
+        // int in its constructor.
+        final int maxMemory = (int) (Runtime.getRuntime().maxMemory() / 1024);
```

Comment:
Fixes https://github.com/openhab/openhab-android/issues/546 ?
I couldn't reproduce the crash with this branch, so I guess this was the reason.

Reply:
This might be the case, but I never saw a OOME myself, so I can't tell personally. AFAICT WebImageCache used a map of unbounded size and at the same time used SoftReferences which are said to not work well on Android (even though I'm not sire whether they're freed to early or too late), so this in any case should be suited better for an Android app.